builder/dockerfile/internals.go

```diff
@@ -544,15 +544,27 @@ func (b *Builder) run(cID string) (err error) {
 	}()
 
 	if err := b.docker.ContainerStart(cID, nil, true, ""); err != nil {
+		if cancelErr := <-cancelErrCh; cancelErr != nil {
+			logrus.Debugf("Build cancelled (%v) and got an error from ContainerStart: %v",
+				cancelErr, err)
+		}
 		return err
 	}
 
 	// Block on reading output from container, stop on err or chan closed
 	if err := <-errCh; err != nil {
+		if cancelErr := <-cancelErrCh; cancelErr != nil {
+			logrus.Debugf("Build cancelled (%v) and got an error from errCh: %v",
+				cancelErr, err)
+		}
 		return err
 	}
 
 	if ret, _ := b.docker.ContainerWait(cID, -1); ret != 0 {
+		if cancelErr := <-cancelErrCh; cancelErr != nil {
+			logrus.Debugf("Build cancelled (%v) and got a non-zero code from ContainerWait: %d",
```

Comment:
In the test you always see `Build cancelled (build cancelled) and got a non-zero code from ContainerWait: 137`